part-2.py

```diff
@@ -4,13 +4,39 @@
 # the appropriate comment.
 
 # search
+def search(array, query):
+    # base case
+    if not array: 
+        return False
+    if array.pop() == query:
+        return True
 
+    return search(array[:-1], query)
 
 
 # is_palindrome
-
+def is_palindrome(text):
+    if len(text) == 1:
+        return True
+    if text[0] == text[-1]:
+        return is_palindrome(text[1:-1]) 
+    else:
+        return False
 
 
 # digit_match
-
+def digit_match(num1, num2):
+    #base case
+    if num1 == num2:
+        return 1
```

Comment:
This solution is almost there, but what if the nums were the same but were longer than one digit?